dashboard/lib/services/lti.rb

```diff
@@ -192,6 +192,11 @@ def self.sync_section_roster(lti_integration, lti_section, nrps_section)
         user_was_new = user.new_record?
         had_changes ||= (user_was_new || user.changed?)
         user.save!
+        lti_user_identity = Queries::Lti.lti_user_identity(user, lti_integration)
+        deployment = lti_section.lti_course&.lti_deployment
+        unless deployment.lti_user_identities.include?(lti_user_identity) || deployment.nil?
```

Comment:
The `deployment.nil?` check seems unnecessary here since if it's nil, then by definition it won't include the lti_user_identity. Could also do `deployment&.lti_user_identities&.include?(lti_user_identity)` to be safe in case deployment is nil.